compute/instances/creating-instance-with-custom-machine-type/custom_machine_type_helper.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
```

Comment:
The encoding of Custom Machine Type restrictions is a helpful demonstration for customers.
What happens when those machine constraints are changed?
IMO, this code would be more beneficial as a library, or an addition to the compute client library.